File: Proyecto - Mysql/DataAccess/Repositories/MasterRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MySql.Data.MySqlClient;

namespace DataAccess.Repositories
{
    public abstract class MasterRepository : Repository
    {
        protected List<MySqlParameter> parameters;

        protected int ExecuteNonQuery(string transactSql)
        {
            using (var connection = GetConnection())
            {
                connection.Open();
                using (var command = new MySqlCommand())
                {
                    command.Connection = connection;
                    command.CommandText = transactSql;
                    command.CommandType = CommandType.Text;
                    foreach (MySqlParameter item in parameters)
                    {
                        command.Parameters.Add(item);
                    }
                    int result = command.ExecuteNonQuery();
                    parameters.Clear();
                    return result;
                }

            }

        }

        protected DataTable ExecuteReader(string transactSql)
        {
            using (var connection = GetConnection())
            {
                connection.Open();
                using (var command = new MySqlCommand())
                {
                    command.Connection = connection;
                    command.CommandText = transactSql;
                    command.CommandType = CommandType.Text;
                    MySqlDataReader reader = command.ExecuteReader();
                    using (var table = new DataTable())
                    {
                        table.Load(reader);
                        reader.Dispose();
                        return table;
                    }
                }

            }
        }

    }
}

File: CrudLayered_VB - MySql/DataAccess/Repositories/MasterRepository.vb
Imports MySql.Data.MySqlClient

Public Class MasterRepository
    Inherits Repository

    Protected parameters As List(Of MySqlParameter)

    Protected Function ExecuteNonQuery(transactSql As String) As Integer
        Using connection = GetConnection()
            connection.Open()
            Using Command = New MySqlCommand()
                Command.Connection = connection
                Command.CommandText = transactSql
                Command.CommandType = CommandType.Text
                For Each item As MySqlParameter In parameters
                    Command.Parameters.Add(item)
                Next
                Dim result = Command.ExecuteNonQuery()
                parameters.Clear()
                Return result
            End Using
        End Using
    End Function

    Protected Function ExecuteReader(transactSql As String) As DataTable
        Using connection = GetConnection()
            connection.Open()
            Using Command = New MySqlCommand()
                Command.Connection = connection
                Command.CommandText = transactSql
                Command.CommandType = CommandType.Text
                Dim reader = Command.ExecuteReader()
                Using table = New DataTable()
                    table.Load(reader)
                    reader.Dispose()
                    Return table
                End Using
            End Using
        End Using
    End Function
End Class
